app/controllers/assignments_controller.rb

```diff
@@ -309,6 +309,38 @@ def batch_runs
     end
   end
 
+  # Return the chart data of assignment grade distributions
+  def grade_distribution_graph_data
+    assignment = Assignment.find(params[:id])
+    labels = []
+    (0..19).each { |i| labels.push((5 * i).to_s + '-' + (5 * i + 5).to_s) }
+    datasets = [
+      {
+        data: assignment.grade_distribution_array,
+        backgroundColor: [
+          'rgba(36, 81, 133, 1)'
+        ],
+        borderColor: [
+          'rgba(36, 81, 133, 1)'
+        ],
+        borderWidth: 1
+      }
+    ]
+    data = { labels: labels, datasets: datasets }
+    options = {
```

Comment:
This is a valid approach, but I'd rather keep these "display" options in the front-end. So what you should do is define these options in the React code, rather than in this method.